### ISSUE TITLE
OSMemoryBarrier() deprecated in OSX 10.12

### ISSUE THREAD

**Issue by ahxxm** [google/leveldb#415]
compiler suggests to use std::atomic_thread_fence() instead:

```
In file included from ./port/port_posix.h:47:
./port/atomic_pointer.h:59:3: warning: 'OSMemoryBarrier' is deprecated: first deprecated in macOS 10.12 - Use
      std::atomic_thread_fence() from <atomic> instead [-Wdeprecated-declarations]
  OSMemoryBarrier();
  ^
/usr/include/libkern/OSAtomicDeprecated.h:749:9: note: 'OSMemoryBarrier' has been explicitly marked deprecated here
void    OSMemoryBarrier( void );
        ^
1 warning generated.
```

**Comment by droark** [comment]
I [filed a patch](https://github.com/google/leveldb/pull/422) for this warning.

**Comment by pwnall** [comment]
Fixed in 594cc987af2e0af6417c4ac2b947ee8cdad59e5e. Sorry this took a while!
